### ISSUE TITLE
admin API by default listening also on IPv6 not working in IPv4 only GKE cluster

### ISSUE THREAD

**Issue by dandynaufaldi** [Kong/charts#1073]
Per the changes done on https://github.com/Kong/charts/pull/986 admin API now by default will listen on both `0.0.0.0` (IPv4) and `[::]` (IPv6)

This default settings doesn't work well with cluster running IPv4 only stack. We got following error from the kong proxy container

```
proxy nginx: [emerg] bind() to [::1]:8444 failed (99: Cannot assign requested address)
```

### helm values

```
admin:
  enabled: false
```

*basically the same with the default values

#### Cluster

- k8s: 1.29.1
- network stack type: IPv4

#### Kong

- chart version: 2.38.0

**Comment by dandynaufaldi** [comment]
btw if anyone stumbled upon similar error (`proxy nginx: [emerg] bind() to [::1]:8444 failed (99: Cannot assign requested address)`)

you can add following env variable in the Helm values to override the Kong Admin listen config so that it won't try to bind on IPv6 address and use the IPv4 only

```
env:
  ADMIN_LISTEN: 0.0.0.0:8444 http2 ssl
```

**Comment by mpdafa** [comment]
Finally, thanks mas! several hours looking for the solution, even gpt wont help haha

**Comment by bluemalkin** [comment]
Has anyone managed to get Kong working on GKE without the admin API enabled at all ? I get this error message:

```
Error: unable to build kong api client(s): creating Kong client: creating Kong client: parsing URL: parse "http://localhost:9999 # You have no admin listens! The controller will not work unless you set .Values.admin.http.enabled=true or .Values.admin.tls.enabled=true!": invalid port ":9999 #
```

I am not sure why the chart `v2.47.0` wants admin API enabled by default ? I've never had it enabled with older versions before (`v2.26.4`).

**Comment by dandynaufaldi** [comment]
Hi @bluemalkin could you share other values you've supplied to the chart?

based on the error message itself, it's form this line https://github.com/Kong/charts/blob/kong-2.47.0/charts/kong/templates/_helpers.tpl#L434

```
{{/*
Return the local admin API URL, preferring HTTPS if available
*/}}
{{- define "kong.adminLocalURL" -}}
  {{- if .Values.admin.tls.enabled -}}
https://localhost:{{ .Values.admin.tls.containerPort }}
  {{- else if .Values.admin.http.enabled -}}
http://localhost:{{ .Values.admin.http.containerPort }}
  {{- else -}}
http://localhost:9999 # You have no admin listens! The controller will not work unless you set .Values.admin.http.enabled=true or .Values.admin.tls.enabled=true!
  {{- end -}}
{{- end -}}
```

by any chance you've set `.Values.admin.tls.enabled` as `false` ?

**Comment by bluemalkin** [comment]
> by any chance you've set `.Values.admin.tls.enabled` as `false` ?

@dandynaufaldi Yes I did. Thanks, I just realised it all works if I leave `.Values.admin.tls.enabled` as  default `true` and `.Values.admin.enabled` as default `false` and update the env `ADMIN_LISTEN`  

It's kind of confusing requiring a value `true` under the `admin` which is not enabled.
